lib/src/arithmetic/signals/fixed_point_logic.dart

```diff
@@ -7,61 +7,93 @@
 // 2024 October 24
 // Author: Soner Yaldiz <soner.yaldiz@intel.com>
 
-import 'dart:math';
+import 'package:meta/meta.dart';
 import 'package:rohd/rohd.dart';
 import 'package:rohd_hcl/rohd_hcl.dart';
 
 /// A representation of (un)signed fixed-point logic following
 /// Q notation (Qm.n format) as introduced by
 /// (Texas Instruments)[https://www.ti.com/lit/ug/spru565b/spru565b.pdf].
-class FixedPoint extends Logic {
-  /// [signed] indicates whether the representation is signed.
-  final bool signed;
+class FixedPoint extends LogicStructure {
+  /// The integer part of the fixed-point number.
+  Logic integer;
 
-  /// [m] is the number of bits reserved for the integer part.
-  final int m;
+  /// The fractional part of the fixed-point number.
+  Logic fraction;
 
-  /// [n] is the number of bits reserved for the fractional part.
-  final int n;
+  /// Return true if signed
+  bool get signed => _signed;
 
-  static int _fixedPointWidth(bool s, int a, int b) => s ? 1 + a + b : a + b;
+  late final bool _signed;
 
-  /// Constructs a [FixedPoint] signal.
-  FixedPoint(
-      {required this.signed,
-      required this.m,
-      required this.n,
-      super.name,
-      super.naming})
-      : super(width: _fixedPointWidth(signed, m, n)) {
-    if ((m < 0) | (n < 0)) {
-      throw RohdHclException('m and n must be non-negative');
-    }
-    if (max(m, n) == 0) {
-      throw RohdHclException('either m or n must be greater than zero');
+  /// [mWidth] is the number of bits reserved for the integer part.
+  int get mWidth => integer.width - (signed ? 1 : 0);
+
+  /// [nWidth] is the number of bits reserved for the fractional part.
+  int get nWidth => fraction.width;
+
+  /// Utility to keep track of the Logic structure name by attaching it
+  /// to the Logic signal name in the output Verilog.
+  static String _nameJoin(String? structName, String signalName) {
+    if (structName == null) {
+      return signalName;
     }
+    return '${structName}_$signalName';
   }
 
+  /// Constructs a [FixedPoint] signal.
+  FixedPoint(
+      {required int mWidth,
+      required int nWidth,
+      bool signed = true,
+      String? name})
+      : this._(
+            Logic(
+                width: mWidth + (signed ? 1 : 0),
+                name: _nameJoin(name, 'integer'),
```

Comment:
is this a good pattern to prepend names of signals inside of structs with the struct name?  this sort of feels like we should more generally solve it as part of ROHD rather than expect everyone making structures to remember to prepend a struct name on the signals.

Reply:
It's not a great pattern:  it seems like we need something in the LogicStructure constructor to name the fields.

```dart
LogicStructure(Iterable<Logic> elements, {String? name})
````
But the problem is that elements can be LogicStructures as well, and so we would need an equivalent naming structure to match the LogicStructure hierarchy.  Maybe that isn't too hard to do statically [name, [name, name, name], name] would be three fields with the middle being another LogicStructure with a set of three Logics.

Reply:
I was thinking more like:
- the `toString()` method reveals information about `parentStructure` for debug
- the SV generator can prepend names of `parentStructure` to generated signals

Thus you'd just be able to name things normally and everywhere you see the name it would be apparent where it was located.  Any problems with that that I'm missing?